### ISSUE TITLE
 in child_ai.py hard to understand the curiosity_behavior function

### ISSUE THREAD

**Issue by aymeric75** [trueagi-io/hyperon-experimental#449]
Hi,

I am posting here, because I don't think the **child_ai** repo is maintained, sorry for the duplicate

So, from **child_ai.py**

we have 

```

# Initial data preparation
data_motivation = {
    'Time': [1], 
	'Motivation': ['Reflex'],
	'Need': ['Pay Attention on Unusual Event'],
	'Action': ['Research Unusual Event'],
    'Result': ['Understand Unusual Event'], 
    'Status': [[['Visual', 'Other'],['Hearing', 'Other'],['Touch', 'Other']]]
    }
Motivation_Global = pd.DataFrame(data_motivation)
```

then

```

# Business logic of curiosity behaviour of Child
def curiosity_behavior(Start, End):
    for t in range(Start, End+1):
        Motivation_item = Motivation_Global[Motivation_Global['Time']==t]
        if Motivation_item.size != 0:
            Is_Result_Achieved = PDCA_func(Motivation_item)
            if Is_Result_Achieved:
                print("Result is achieved! Time: ", t)
            t = t + 1
    print("curiosity_behavior ends.")
```

`Motivation_Global['Time']` is equal to `[1]`  , so basically `[1]` will never be equal to 1 or 2 or 3 etc.  that's why I don't understand this line:
`
Motivation_item = Motivation_Global[Motivation_Global['Time']==t]`

Furthermore, `Motivation_Global['Time']==t` will return either True or False, but `Motivation_Global[True]` or `Motivation_Global[False] `does not exists...

Thanks

**Comment by aymeric75** [comment]
> @aymeric75 , indeed, metta_examples repo is in general intended for potentially not maintained examples, which might still be interesting, but which we don't want to put into the main repo exactly because they are potentially not maintained. And this seems to be the case for child_ai. If you are still interested in this example, you can try contacting the author directly.

thanks, I wanted to test child ai because I have been adviced to do so in order to understand better MeTTa from an example, in this case, do you know a better example that is maintained and that is also representative of MeTTa (in the sense that it can display something """close""" to AGI, like a child ai..).

**Comment by Necr0x0Der** [comment]
As far as I remember, Child AI tried to use MeTTa on the level of atomspace only, so it is not a good example. For now, the most advanced example of trying to code AGI-ish components in MeTTa is https://github.com/trueagi-io/hyperon-pln . Will it work for you?

**Comment by aymeric75** [comment]
Thanks but I am not sure it displays all the possibilities of MeTTa, what I see in this repo are more tools/functions than concrete full fletched examples, for instance the DeductionDTLTest.metta...

**Comment by Necr0x0Der** [comment]
> I am not sure it displays all the possibilities of MeTTa

Yes, it doesn't. We have a goal to prepare other advanced examples covering different aspects of MeTTa in few months.

**Comment by aymeric75** [comment]
So if I understand correctly, OpenCog / MeTTa is the only opensource project to build an AGI out there, but it's still hard to have a grasp on it since no concrete example is given

**Comment by aymeric75** [comment]
or, to be fair... it's hard to have a grasp without such examples but still possible, it only requires much more time
